### PR TITLE
[APPC-1449] Fixed wrong wallet verification status when no internet connection

### PR DESCRIPTION
**Pull request by carvalho2707**
**What does this PR do?**

   This PR fixed an use case where the settings screen would show a wrong wallet validation status if no network connection is available

**Database changed?**

  No

**Where should the reviewer start?**

- [ ] SettingsFragment.kt

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [APPC-1449](https://aptoide.atlassian.net/browse/APPC-1449)

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-1449](https://aptoide.atlassian.net/browse/APPC-1449)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/com/asfoundation/wallet/interact/SmsValidationInteract.kt [hunk 1]
```diff
@@ -20,6 +23,19 @@ class SmsValidationInteract(
...
                    code: String): Single<WalletValidationStatus> {
     return smsValidationRepository.validateCode(phoneNumber, wallet.address, code)
+        .doOnSuccess { saveWalletVerifiedStatus(it, wallet.address) }
+  }
+
+  private fun saveWalletVerifiedStatus(status: WalletValidationStatus, walletAddress: String) {
+    if (status == WalletValidationStatus.DOUBLE_SPENT || status == WalletValidationStatus.SUCCESS) {
+      defaultSharedPreferences.edit()
+          .putBoolean(WALLET_VERIFIED + walletAddress, true)
+          .apply()
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/interact/SmsValidationInteract.kt]: This logic should be inside the sharedPreference. The only thing that should be here is the if

File: app/src/main/java/com/asfoundation/wallet/interact/SmsValidationInteract.kt [hunk 2]
```diff
@@ -20,6 +23,19 @@ class SmsValidationInteract(
...
+    }
+  }
+
+  private companion object {
+    private const val WALLET_VERIFIED = "wallet_verified_"
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/interact/SmsValidationInteract.kt]: This should also go to the shared preferences

File: app/src/main/java/com/asfoundation/wallet/ui/SettingsPresenter.kt [hunk 3]
```diff
@@ -66,5 +77,10 @@ class SettingsPresenter(private val view: SettingsView,
           view.setRedeemCodePreference(wallet.address)
         })
   }
+
+  private companion object {
+    private const val PREF_WALLET = "pref_wallet"
+    private const val WALLET_VERIFIED = "wallet_verified_"
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/ui/SettingsPresenter.kt]: This should be in the defaulsSharedPrefs

**Review by JDSM01 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/asfoundation/wallet/di/AccountsManageModule.java [hunk 4]
```diff
@@ -34,8 +35,9 @@
   }
 
   @Provides DeleteWalletInteract provideDeleteAccountInteract(
-      WalletRepositoryType accountRepository, PasswordStore store) {
-    return new DeleteWalletInteract(accountRepository, store);
+      WalletRepositoryType accountRepository, PasswordStore store,
+      PreferenceRepositoryType preferenceRepositoryType) {
+    return new DeleteWalletInteract(accountRepository, store, preferenceRepositoryType);
```
Comment (JoaoRaimundo) [inline app/src/main/java/com/asfoundation/wallet/di/AccountsManageModule.java]: Is the preference missing an 's' a typo?

**Review by JDSM01 — APPROVED** [review]

**Pull request merged**